packages/gatsby/src/utils/page-data.ts

```diff
@@ -0,0 +1,79 @@
+import fs from "fs-extra"
+import path from "path"
+import { IGatsbyPage } from "../redux/types"
+import { store } from "../redux"
+
+interface IPageDataContext {
+  publicDir: string
+}
+
+interface IPageData
+  extends Pick<IGatsbyPage, "componentChunkName" | "matchPath" | "path"> {
+  result: any
+}
```

Comment:
this is a bit of a nit. But I find these utility generics hard to understand. Can we change this to this?

```ts
interface IPageData {
  componentChunkName: IGatsbyPage["componentChunkName"]
  matchPath: IGatsbyPage["matchPath"]
  path: IGatsbyPage["path"]
  result: any // also, can this be improved?
}

Reply:
I think `any` might be an instance of `ExecutionResult` from graphql. Try that out